pkg/client/client.go

```diff
@@ -25,6 +25,8 @@ package client
 // The connection may consume expensive resources, like TCP connections, or file descriptors, so it
 // is important to reuse it as much as possible, and to close it once it is no longer needed.
 type Connection interface {
+	// Connect creates a ne connection to the messaging broker.
+	Connect() error
```

Comment:
How will the caller specify the connection parameters? Can we have something like this:

```go
connection, err := client.NewConnectionBuilder().
  Host("myhost").
  Port(123).
  User("myuser").
  Password("mypass").
  Build()
if err != nil {
  ...
}
```

Reply:
I don't mind though it's very java like. We can also start with a static struct init which is more "GOish"

```
connection := StompConnection{
  brokerHost: "myhost",
  brokerPort: 123,
  user: "myuser",
  password: "mypass",
}

err := connection.Connect()
if err != nil {
  ...
}